public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -65,27 +108,12 @@ export const getMatcherQueryParams = (labels: Labels) => {
  * this function will normalize all of the different ways to define matchers in to a single one.
  */
 export const normalizeMatchers = (route: Route): ObjectMatcher[] => {
-  const matchers: ObjectMatcher[] = [];
+  let matchers: ObjectMatcher[] = [];
 
   if (route.matchers) {
     route.matchers.forEach((matcher) => {
-      const { name, value, isEqual, isRegex } = parseMatcher(matcher);
-      let operator = MatcherOperator.equal;
-
-      if (isEqual && isRegex) {
-        operator = MatcherOperator.regex;
-      }
-      if (!isEqual && isRegex) {
-        operator = MatcherOperator.notRegex;
-      }
-      if (isEqual && !isRegex) {
-        operator = MatcherOperator.equal;
-      }
-      if (!isEqual && !isRegex) {
-        operator = MatcherOperator.notEqual;
-      }
-
-      matchers.push([name, operator, value]);
+      const parsedMatchers = parseMatcherToArray(matcher).map(matcherToObjectMatcher);
+      matchers = matchers.concat(parsedMatchers);
```

Comment:
The fact that single `matcher` can be parsed to multiple `parsedMatchers` is baffling but I don't have any idea on how to simplify that 😅